sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -269,7 +284,18 @@ public Path getPath(URI uri) {
     @Override
     public SeekableByteChannel newByteChannel(Path path, Set<? extends OpenOption> set,
             FileAttribute<?>... fileAttributes) throws IOException {
-        throw LoggingUtility.logError(logger, new UnsupportedOperationException());
+        if (Objects.isNull(set)) {
+            set = Collections.emptySet();
+        }
+
+        if (set.contains(StandardOpenOption.WRITE)) {
```

Comment:
Should we warn users if they set both read and write?

Reply:
It'll actually throw if they do. Because Read is not a supported option on outputstream and vice versa.